atc/worker/pool.go

```diff
@@ -277,6 +279,13 @@ func (pool *pool) SelectWorker(
 ) (Client, time.Duration, error) {
 	logger := lagerctx.FromContext(ctx)
 
+	if callbacks != nil {
+		err := callbacks.BeforeSelectWorker(logger)
```

Comment:
I'd rather have this (and the `SelectedWorker` details) in the exec package rather than via these callbacks

Reply:
@aoldershaw I don't quite get your point, can you please clarify?